### DISCUSSION TITLE
Minor Issue with Catching Empty Fields

### DISCUSSION THREAD

**Discussion by JEB-24** [PHPMailer/PHPMailer#3002] [General]
I'm very new to this I'm afraid. Please forgive my ignorance.

I've managed to hack together my code for PHPMailer from various sources online and it is working. I'm not using Composer. It works and emails are being sent. Unfortunately, I'm having a minor issue with catching exceptions. I have a simple contact form with three fields. The success message works. I would like an error message to occur if any of the fields are left blank. At the moment the error message only occurs if the message field is empty. I would really like the error message to be triggered when any of the fields are empty and the submit button is clicked.

Here is my PHP code (I've removed the mailbox credentials):

`<?php

    use PHPMailer\PHPMailer\PHPMailer;
    use PHPMailer\PHPMailer\SMTP;
    use PHPMailer\PHPMailer\Exception;
	
    require './src/Exception.php';
    require './src/PHPMailer.php';
    require './src/SMTP.php';
	
	
    $myPersonalEmail = "";   
    $externalMailHost = "";
    $externalMailAddress = "";
    $externalMailSMTPAuth = false;
    $externalMailUsername = "";
    $externalMailPassword = "";
    $externalMailSMTPSecure = "tls";
    $externalMailPort = 587;
    
	
    if(isset($_POST['submit'])) {

        $mail = new PHPMailer(true);

        $mail->SMTPDebug = 0;
        $mail->isSMTP();

        $mail->Host = $externalMailHost;
        $mail->SMTPAuth = $externalMailSMTPAuth;
        $mail->Username = $externalMailUsername;
        $mail->Password = $externalMailPassword;
        $mail->SMTPSecure = $externalMailSMTPSecure;
        $mail->Port = $externalMailPort;
        
        $mail->setFrom($externalMailAddress, 'Mailer');
        $mail->addAddress($myPersonalEmail);
        $mail->addReplyTo($_POST['email'], $_POST['name']);

        $mail->isHTML(true);    
        $mail->Subject = 'D&H Website Contact Form';
        $mail->Body = $_POST['message'];

        try {
            $mail->send();
            
			echo
			
			"
			<script>
				alert('Thank you, your message was sent successfully');
				document.location.href = 'index.html';	
			</script>
			";
			
				} catch (Exception $e) {
					
					echo 
			
					"
					<script>
						alert('Your message could not be sent please ensure all fields are complete.');
						document.location.href = 'index.html#contact';	
					</script>
					";}
			} 	
?>`

I would be grateful for any assistance or guidance. 

Thanks

John

**Comment by JEB-24** [comment]
I made some changes to the code and it is now behaving as I expected.

**Reply by Synchro** [reply]
Glad you figured it out. Generally speaking, it's not a PHPMailer issue, but more about the context you're using it from, so just as you check for form submission with `if(isset($_POST['submit'])) {`, you can check that other values are set before you start doing anything involving sending email. For example (this is a bit blunt!):

```php
if (!isset($_POST['name'])) {
    echo 'Fill in the name field';
    exit;
}
```

**Reply by JEB-24** [reply]
Thank you very much for your suggestion. I give this a try.
